### ISSUE TITLE
commit: it's not caching a missing output

### ISSUE THREAD

**Issue by jorgeorpinel** [iterative/dvc#5034]
## Bug Report

Start from a fresh DVC repo (no .dvc/cache dir exists)

```console
λ dvc run -n hello -d hello --no-exec "echo hi > hello"
...
λ ls
dvc.yaml
λ cat dvc.yaml
stages:
  hello:
    cmd: echo hi > hello
    deps:
    - hello
λ  echo hi > hello
λ ls
dvc.lock  hello

λ dvc commit
dependencies ['hello'] of stage: 'hello' changed. Are you sure you want to commit it? [y/n] y
Generating lock file 'dvc.lock'
Updating lock file 'dvc.lock'

λ ls
dvc.lock  dvc.yaml  hello
λ cat dvc.lock
hello:
  cmd: echo hi > hello
  deps:
  - path: hello
    md5: 764efa883dda1e11db47671c4a3bbd9e
    size: 3
```

Everything looks good from the terminal output, but still there's no .dvc/cache dir. `hello` is not cached, nor a link.

### Please provide information about your setup

**Output of `dvc version`:**

```console
λ dvc version
DVC version: 1.10.2 (exe)
---------------------------------
Platform: Python 3.7.9 on Windows-10-10.0.18362-SP0
Supports: All remotes
Cache types: hardlink
Cache directory: NTFS on C:\
Caches: local
Remotes: None
Workspace directory: NTFS on C:\
Repo: dvc, git```

**Additional Information (if any):**

If applicable, please also provide a `--verbose` output of the command, eg: `dvc add --verbose`.

**Comment by jorgeorpinel** [comment]
Oops wait, I submitted this before finishing the description accidentally... ⌛

**Comment by jorgeorpinel** [comment]
OK, ready, please look at the updated description directly on GH.

**Comment by jorgeorpinel** [comment]
> λ dvc commit
dependencies ['hello'] of stage: 'hello' changed. Are you sure you want to commit it? [y/n] y

p.s. is there a way to run commit without interaction? For scripts.

**Comment by skshetry** [comment]
@jorgeorpinel, I think this happens because you added it as a `-d hello` rather than a output.
We don't track dependencies at all unless they are output on some other stages.

**Comment by jorgeorpinel** [comment]
Oops true 🤦 let me try again...

**Comment by jorgeorpinel** [comment]
Yup, my bad. Typo in `run`. Nvmd!
